snuba/utils/streams/abstract.py

```diff
@@ -72,6 +72,24 @@ class Consumer(ABC, Generic[TStream, TOffset, TValue]):
     An abstract class that provides methods to consume messages from a
     multiplexed collection of streams.
 
+    Streams support sequential access, as well as random access by offsets.
+    The specific data types of offsets are implementation dependent, but
+    offset values should represent a totally-ordered, monotonic sequence.
```

Comment:
does `totally-ordered + monotonic` actually imply  `strictly monotonic` ? (to exclude duplicate offsets ? Or should that be `strictly ordered + monotonic`?

Reply:
I think either of "strictly monotonic" or "strict total order" makes sense here though I'm not sure if I am using these 100% literally correctly to be honest.